### PR TITLE
Filter WMS capabilities with correct OGC server

### PR DESCRIPTION
**Pull request by ger-benjamin**
Part of #2929 

TODO:
 - [x] Fix DetachedInstanceError: Instance <LayerWMS at 0x7feae2836290> is not bound to a Session; ~~attribute refresh operation cannot proceed -> maybe come frome the `cache_on_argument` It seems to work with a single argument but not with an array  ?~~ -> must to expung objects in session
 - [x] add tests

(I've not tested the things about doing a curl to get capabilities with basic auth.)

### CODE REVIEW

**Comment by sbrunner** [comment]
You should also add a test case :-)

File: c2cgeoportal/lib/filter_capabilities.py [hunk 1]
```diff
@@ -152,15 +161,19 @@ def caching_prepare_input_source(source, base=None):
     saxutils.prepare_input_source = caching_prepare_input_source
 
 
-def filter_capabilities(content, role_id, wms, wms_url, headers, proxies):
+def filter_capabilities(content, role_id, wms, url, headers, proxies, request, ):
```
Comment (ger-benjamin) [inline c2cgeoportal/lib/filter_capabilities.py]: last `, ` works but looks strange

File: c2cgeoportal/tests/functional/__init__.py [hunk 2]
```diff
@@ -63,65 +64,65 @@
 caching.init_region({"backend": "dogpile.cache.memory"})
 
 
+def cleanup_db():
```
Comment (ger-benjamin) [inline c2cgeoportal/tests/functional/__init__.py]: That clean really all ?  What about metadata, tsearch... ? Perhaps layer ? (or that comes with treeitem ? but both layerv1 and v2 ?)

Comment (sbrunner) [inline c2cgeoportal/tests/functional/__init__.py]: TreeItem cover all the tree, FullTextSearch and Shorturl will be added.

File: c2cgeoportal/lib/filter_capabilities.py (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/lib/filter_capabilities.py]: fixed :-)

**Comment by ger-benjamin** [comment]
Can I merge ?

**Pull request merged**